main.cpp

```diff
@@ -1,478 +1,458 @@
+    // 2) heat up food
+    void heatFood (bool useCookTop = true, int numOfMins = 12);
+    // 3) boil water
+    int timeNeededToBoilWater (int waterInLiters, bool coldWater);
 
+    myRange myFirstRange;
+};
 /*
-===================
-Part 1d assignment: 3 Parts.
-===================
-
-1) Fill in #5 - #9 below with plain-english UDTs for the 5 
-properties you created for UDT #10. 
-    example: 
-        If #10's first property was 'Engine', then `Thing 5)` 
-        will be `Engine`. 
-        You will need to provide 5 properties and 3 member 
-        functions of that Engine object in plain English.
-        Remember to pick properties that can be represented 
-        with 'int float double bool char std::string'.
-
-2) write the name of the primitive type you'll be using after 
-each property for UDTs 5 - 9.
-    You already did this for UDTs 1-4 in Part 1c.
-    Pick properties that can be represented with 
-    'int float double bool char std::string'
-    example: 
-        Display:
-            Number of Pixels (int)
-            Amount of Power consumed (milliwatt-hours) (float)
-            Brightness (double)
-            width in cm (int)
-            height in cm (int)
-
-Step 3 is written below UDTs 5 - 9
-*/
+Thing 3) shopping cart
+ */
+struct shoppingCart 
+{
+    // 1) its material (std::string)
+    std::string cartMaterial = "aluminum";
+    // 2) number of wheels (int)
+    int numWheels = 4;
+    // 3) color of its handle (std::string)
+    std::string colorHandle = "red";
+    // 4) its width (float)
+    float cartWidth = 24.1f;
+    // 5) which store it belongs to (std::string)
+    std::string cartStore = "Albertsons";
+
+    // 1) carries groceries 
+    void carryGrocery (bool badWheels = false, float weightLimit =  40.2f);
+    // 2) it can roll down a hill
+    void cartRollsDown (bool steepHill, float velocity = 30.2f);
+    // 3) can be parked in a cart corral 
+    bool cartParked (std::string cartLocation = "corral", bool cartMoves = false);
+};
 
 /*
-Thing 5)
-5 properties:
-    1)
-    2)
-    3)
-    4)
-    5)
-3 things it can do:
-    1)
-    2)
-    3)
+Thing 4) wind
  */
+struct wind
+{
+    // 1) speed (float)
+    double windSpeed = 3.4234898;
+    // 2) direction (std::string)
+    std::string windDirection = "North";
+    // 3) its definition (std:: string)
+    std::string windDefinition = "movement in air molecules";
+    // 4) its cause (std:: string)
+    std::string windCause = "difference in gas density";
+    // 5) its use (std:: string)
+    std::string windUse = "wind energy";
+
+    // 1) it transports seeds
+     void windTransportSeeds (int numSeed = 4, float windGust = 3.42f);
+    // 2) it turns wind turbines 
+     void windTurnTurbines (std::string windDirection, bool turnTurbine = true, float windGust = 34.2f);
+    // 3) it helps birds travel 
+     void windHelpBirdTravel (std::string windDirection, bool birdFly, float windGust = 34.2f);
```

Comment:
Don't use `wind` in a function name, in a class that has the word `wind` in the name
- [ ] revise